django/db/backends/postgresql/features.py

```diff
@@ -67,3 +67,24 @@ def is_postgresql_10(self):
     has_brin_autosummarize = property(operator.attrgetter('is_postgresql_10'))
     has_phraseto_tsquery = property(operator.attrgetter('is_postgresql_9_6'))
     supports_table_partitions = property(operator.attrgetter('is_postgresql_10'))
+
+    # Views do not support the allows_group_by_selected_pks optimization"
+    def allows_group_by_selected_pks_on_model(self, model):
+        if model._meta.managed:  # We are assuming managed tables are not views, since Django can't create views.
+            return True
+
+        try:
+            return not model._meta._is_view
```

Comment:
If we want to add a new `_is_view` option to `Meta` it should be documented, but I don't think it is necessary.

Reply:
The value is used as a cache for the query so that it isn't done for every access, see line 89. I can switch to the introspecition + lru_cache approach if that's better, but I'm not sure how to mock that value in tests.

Reply:
OK, makes sense. Name is quite misleading because we're checking if it is a table, please rename `_is_view` to `_is_table`. Also, IMO we still should add `_is_table()` (without caching) hook to the introspection.